dev-cpp/waylandpp/waylandpp-0.1.0.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit multilib scons-utils toolchain-funcs versionator
+
+DESCRIPTION="Wayland C++ bindings"
+HOMEPAGE="https://github.com/NilsBrause/waylandpp"
+
+LICENSE="MIT"
+IUSE="doc"
+SLOT="0/$(get_major_version)"
+
+if [[ ${PV} == *9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/NilsBrause/waylandpp.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/NilsBrause/waylandpp/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/waylandpp-${PV}"
+fi
+
+COMMON_DEPEND="
+	>=dev-libs/wayland-1.11.0
+	media-libs/mesa[wayland]
+"
+DEPEND="${COMMON_DEPEND}
+	doc? (
+		app-doc/doxygen
+		media-gfx/graphviz
+	)
+	"
+RDEPEND="${COMMON_DEPEND}"
+
+src_compile() {
+	CC="$(tc-getCXX)" PKG_CONFIG="$(tc-getPKG_CONFIG)" escons
+	if use doc; then
+		doxygen || die "error making docs"
+	fi
+}
+
+src_install() {
+	PREFIX="${D%/}/usr" scons install
+	# fix multilib-strict QA failures
+	mv "${ED%/}"/usr/{lib,$(get_libdir)} || die
```

Comment:
seriously: here's an idea: tell the moronic upstream to switch to Meson (or CMake as a worse fallback), trying to make SCons do want you want is close to impossible without rewriting the whole build system.

Reply:
@SoapGentoo I hope not every Gentoo developer is as impolite as you.

Reply:
@NilsBrause in general not, but when people volunteer to contribute a better build system because SCons is a load of garbage and the upstream is adamant to actually see why SCons is crap, then my tongue-in-cheek comment is meant in less of an exaggerating tone and more in a serious one. But sure, hardcoding assumptions about the filesystem is absolutely fine :+1:

Reply:
@SoapGentoo SCons works perfectly well on every computer I tested it on (each running a different distro). If you have problems using it, you are doing something wrong and should rtfm.
The only assumption on the filesystem I made is the usage of the forward slash and that's perfectly fine because we aren't targeting Microsoft Windows. 
If you are not happy with that, feel free to submit patches for a secondary build system like cmake.